src/components/AuthRequired.js

```diff
@@ -0,0 +1,22 @@
+import React, { Component } from 'react'
+import { Redirect, withRouter } from 'react-router'
+
+class AuthRequired extends Component {
```

Comment:
it does not need to be a full-specced `Component`. Neither lifecycle methods, nor state is used - assuming, it would do nothing more.
In addition, it does that every path change - so this part is executed each time Route change or its parent re-renders, which seems to be redundant job.

On the other hand, it should get the `token` and `user` from `location` and `dispatch` event that user was logged - in this case it will have to be a class to perform this logic.

Reply:
To sum up:
1) outdated because of 3)
2) it should store authorization status in LocalStorage
3) TODO: real authorization with Slack should be made with proper authorization URL and token checking endpoint